### ISSUE TITLE
[Code Generation] Using "value" as an identifier for a component field.

### ISSUE THREAD

**Issue by gabrjt** [spatialos/gdk-for-unity#607]
### Description
The generated code has a bug if you use the identifier ````value```` for a component field.

For example, in ````tick.schema````:
````
package game;

component Tick {
    id = 15000;
    uint32 value = 1;
}
````
Will generate in ````Tick.cs````:
````
private uint value;

public uint Value
{
     get => value;
     set
     {
          MarkDataDirty(0);
          value = value;
     }
}
````
And the private ````value```` field will never be updated.

## Possible solution
Use in the setter the ````this```` keyword, so the generated code will be like:
````
private uint value;

public uint Value
{
     get => value;
     set
     {
          MarkDataDirty(0);
          this.value = value;
     }
}
````
## Steps to reproduce
1. Create a component with a field named value.
2. Regenerate code.
3. Try to update the component Value.
4. It will never be updated.

## Environment
SpatialOS Blank Project 0.1.2 and Unity 2018.2.14f1

**Comment by jamiebrynes7** [comment]
Hey @gabrjt thanks for raising this! We are tracking this internally as UTY-1432

**Comment by jamiebrynes7** [comment]
Opened a PR to fix this - #619

**Comment by jamiebrynes7** [comment]
The PR has been merged 🎉 

This should be fixed in the next release!
